RIOT-OS/RIOT
# Kconfig: Expose gnrc/netif configurations

### Contribution description
This PR moves the configuration macros of `gnrc_netif` found [here](https://github.com/RIOT-OS/RIOT/blob/master/sys/include/net/gnrc/netif/conf.h) to the `CONFIG_` namespace, and exposes them to Kconfig.

As agreed offline with @leandrolanzieri I did not include the default thread priority parameter because this should be defined by the platform, if not changed manually. We might need a better solution to this in future.

Furthermore, I did not include `GNRC_NETIF_NUMOF` because this won't be static anymore in the future.

### Testing procedure
- `examples/gnrc_networking` should still work as usual.
- You should be able to configure the module's parameters via `make menuconfig` in the test application.

### Issues/PRs references
Part of #12888
